wallet/chain/p/builder/builder_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+//nolint:gosec // This file does not need cryptographically strong randomness
+package builder
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+func generateUTXOs(assetID ids.ID, locktime uint64) []*avax.UTXO {
+	utxos := make([]*avax.UTXO, rand.Intn(10))
+	for i := range utxos {
+		var output avax.TransferableOut = &secp256k1fx.TransferOutput{
```

Comment:
(optional, nit)

```go
var output avax.TransferableOut
if lockTime == 0 {
    output = &secp...
} else {
    output = &stakeable...
}
```

I was a little confused as to why you were making it an interface type at the beginning and it only became apparent later that you might overwrite it. The above pattern communicates that.

Reply:
I don't think this is really do-able.

```golang
if locktime != 0 {
	output = &stakeable.LockOut{
		Locktime:        locktime,
		TransferableOut: output,
	}
}
```

references the original `output` (which is set in the body of your `if` statement). So `if` always needs to execute and the `else` only needs to execute sometimes...

Unless the thought would be to have a `secpOutput` variable and then setting `output` to either that or a wrapped version of it... That seems clunkier to me than the current code.

Reply:
Ahh I missed the reference to the original. Sorry.